lib/block_view.go

```diff
@@ -6575,6 +6614,164 @@ func (bav *UtxoView) GetPostsPaginatedForPublicKeyOrderedByTimestamp(publicKey [
 	return postEntries, nil
 }
 
+func (bav *UtxoView) GetDiamondSendersForPostHash(postHash *BlockHash) (_pkidToDiamondLevel map[PKID]int64, _err error) {
+	handle := bav.Handle
+	dbPrefix := append([]byte{}, _PrefixDiamondedPostHashDiamonderPKIDDiamondLevel...)
+	dbPrefix = append(dbPrefix, postHash[:]...)
+	keysFound, _ := EnumerateKeysForPrefix(handle, dbPrefix)
+
+	// Iterate over all the db keys & values and load them into the view.
+	expectedKeyLength := 1 + HashSizeBytes + btcec.PubKeyBytesLenCompressed + 8
+	for _, key := range keysFound {
+		// Sanity check that this is a reasonable key.
+		if len(key) != expectedKeyLength {
+			return nil, fmt.Errorf("UtxoView.GetDiamondsForPostHash: Invalid key length found: %d", len(key))
+		}
+
+		diamondPostHash := &BlockHash{}
+		copy(diamondPostHash[:], key[1:1+HashSizeBytes])
+		diamondPostEntry := bav.GetPostEntryForPostHash(diamondPostHash)
+		receiverPKIDEntry := bav.GetPKIDForPublicKey(diamondPostEntry.PosterPublicKey)
+
+		senderPKID := &PKID{}
+		copy(senderPKID[:], key[1+HashSizeBytes:])
+
+		diamondKey := &DiamondKey{
+			SenderPKID:      *senderPKID,
+			ReceiverPKID:    *receiverPKIDEntry.PKID,
+			DiamondPostHash: *postHash,
+		}
+
+		loadedEntry := bav.GetDiamondEntryForDiamondKey(diamondKey)
+		_ = loadedEntry
+	}
+
+	// Iterate over the view and create the final map to return.
+	pkidToDiamondLevel := make(map[PKID]int64)
+	for _, diamondEntry := range bav.DiamondKeyToDiamondEntry {
+		if !diamondEntry.isDeleted && reflect.DeepEqual(diamondEntry.DiamondPostHash[:], postHash[:]) {
+			pkidToDiamondLevel[*diamondEntry.SenderPKID] = diamondEntry.DiamondLevel
+		}
+	}
+
+	return pkidToDiamondLevel, nil
+}
+
+func (bav *UtxoView) GetLikesForPostHash(postHash *BlockHash) (_likerPubKeys [][]byte, _err error) {
+	handle := bav.Handle
+	dbPrefix := append([]byte{}, _PrefixLikedPostHashToLikerPubKey...)
+	dbPrefix = append(dbPrefix, postHash[:]...)
+	keysFound, _ := EnumerateKeysForPrefix(handle, dbPrefix)
+
+	// Iterate over all the db keys & values and load them into the view.
+	expectedKeyLength := 1 + HashSizeBytes + btcec.PubKeyBytesLenCompressed
+	for _, key := range keysFound {
+		// Sanity check that this is a reasonable key.
+		if len(key) != expectedKeyLength {
+			return nil, fmt.Errorf("UtxoView.GetLikesForPostHash: Invalid key length found: %d", len(key))
+		}
+
+		likedPostHash := &BlockHash{}
+		copy(likedPostHash[:], key[1:1+HashSizeBytes])
```

Comment:
we know the likedPostHash as it is the input to the function, so we can just copy this slice once before the foor loop